### PR TITLE
POC filters dependencies. 

### PR DESCRIPTION
**Pull request by mahdibenromdhane**
- Passing Issuer filter to Tag filter in users list
- Passing Site filter to SiteArea filter in charging stations list

### CODE REVIEW

File: src/app/pages/charging-stations/list/charging-stations-list-table-data-source.ts [hunk 1]
```diff
@@ -325,11 +325,12 @@ export class ChargingStationsListTableDataSource extends TableDataSource<Chargin
...
   public buildTableFiltersDef(): TableFilterDef[] {
     if (this.isOrganizationComponentActive) {
+      const siteFilter = new SiteTableFilter().getFilterDef();
       return [
         // new ChargingStationTableFilter().getFilterDef(),
         new IssuerFilter().getFilterDef(),
-        new SiteTableFilter().getFilterDef(),
-        new SiteAreaTableFilter().getFilterDef()
+        siteFilter,
+        new SiteAreaTableFilter(null, [siteFilter]).getFilterDef(),
```
Comment (LucasBrazi06) [inline src/app/pages/charging-stations/list/charging-stations-list-table-data-source.ts:333]: sideIDs should be removed from the SiteAreaTableFilter and replaced by the filter deps.

File: src/app/shared/table/filters/site-area-table-filter.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { FilterType, TableFilterDef } from '../../../types/Table';
 import { TableFilter } from './table-filter';
 
 export class SiteAreaTableFilter extends TableFilter {
-  constructor(siteIDs?: ReadonlyArray<string>) {
+  constructor(siteIDs?: ReadonlyArray<string>, dependentFilters?: TableFilterDef[]) {
```
Comment (LucasBrazi06) [inline src/app/shared/table/filters/site-area-table-filter.ts:6]: siteIDs param to remove

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]
The concept is validated!
Check my comments for the rest.

File: src/app/pages/charging-stations/list/charging-stations-list-table-data-source.ts (hunk 1 above)
Comment (mahdibenromdhane) [reply src/app/pages/charging-stations/list/charging-stations-list-table-data-source.ts]: Done

File: src/app/shared/table/filters/site-area-table-filter.ts (hunk 2 above)
Comment (mahdibenromdhane) [reply src/app/shared/table/filters/site-area-table-filter.ts]: Done

**Review by LucasBrazi06 — APPROVED** [review]
Gut!
PoC is validated.
You can generalize the feature to all other filters in all the app.
We have to think also about this use case: when the Site is changed, Site Area filter should be cleared :-)
I'll create a new issue for this.

**Comment by mahdibenromdhane** [comment]
> Gut!
> PoC is validated.
> You can generalize the feature to all other filters in all the app.
> We have to think also about this use case: when the Site is changed, Site Area filter should be cleared :-)
> I'll create a new issue for this.

Great :)

**Pull request merged**